test/test_foreach.py

```diff
@@ -353,105 +339,64 @@ def test_unary_slowpath(self, device, dtype, op):
-        self.assertEqual(expected, res)
-
-        expected = [torch.min(a1, b1) for a1, b1 in zip(a, b)]
-        res = torch._foreach_minimum(a, b)
-        self.assertEqual(expected, res)
-
-    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=True, include_bfloat16=False)))
-    def test_max_min_inf_nan(self, device, dtype):
-        a = [
-            torch.tensor([inf], device=device, dtype=dtype),
-            torch.tensor([-inf], device=device, dtype=dtype),
-            torch.tensor([nan], device=device, dtype=dtype),
-            torch.tensor([nan], device=device, dtype=dtype)
-        ]
-
-        b = [
-            torch.tensor([-inf], device=device, dtype=dtype),
-            torch.tensor([inf], device=device, dtype=dtype),
-            torch.tensor([inf], device=device, dtype=dtype),
-            torch.tensor([nan], device=device, dtype=dtype)
-        ]
-
-        expected_max = [torch.max(a1, b1) for a1, b1 in zip(a, b)]
-        res_max = torch._foreach_maximum(a, b)
-        self.assertEqual(expected_max, res_max)
-
-        expected_min = [torch.min(a1, b1) for a1, b1 in zip(a, b)]
-        res_min = torch._foreach_minimum(a, b)
-        self.assertEqual(expected_min, res_min)
+    @dtypes(*torch.testing.get_all_dtypes(include_half=True, include_bfloat16=True, include_complex=False))
+    @ops(foreach_minmax_op_db)
+    def test_minmax_slowpath(self, device, dtype, op):
+        for N in N_values:
+            inputs = tuple(op.sample_inputs(device, dtype, N, noncontiguous=True) for _ in range(2))
+            self._minmax_test(op, inputs, False, 1)
+
+    # note(mkozuki): ForeachFuncInfo's of both `_foreach_maximum` and `_foreach_minimum` include integer types.
+    # so, manually limit dtypes to fp types for inf&nan tests.
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=True, include_half=True))
+    @ops(foreach_minmax_op_db)
+    def test_minmax_float_inf_nan(self, device, dtype, op):
+        inputs = (
+            [
+                torch.tensor([float('inf')], device=device, dtype=dtype),
+                torch.tensor([-float('inf')], device=device, dtype=dtype),
+                torch.tensor([float('nan')], device=device, dtype=dtype),
+                torch.tensor([float('nan')], device=device, dtype=dtype)
+            ],
+            [
+                torch.tensor([-float('inf')], device=device, dtype=dtype),
+                torch.tensor([float('inf')], device=device, dtype=dtype),
+                torch.tensor([float('inf')], device=device, dtype=dtype),
+                torch.tensor([float('nan')], device=device, dtype=dtype)
+            ],
+        )
+        self._minmax_test(op, inputs, True, 1)
+
+    @ops(foreach_minmax_op_db)
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=True, include_half=True))
+    def test_minmax_inf_nan(self, device, dtype, op):
+        inputs = (
+            [
+                torch.tensor([inf], device=device, dtype=dtype),
```

Comment:
can you please use `float('inf')` etc here, and remove `torch._six` imports? Also, how is this test different from the previous one?

Reply:
> can you please use `float('inf')` etc here, and remove `torch._six` imports?

Does merge of `test_minmax_inf_nan` and `test_minmax_float_inf_nan` sound good to you?



> Also, how is this test different from the previous one?

The only difference is the use of `@ops` decorator.